src/MSBuild/app.amd64.config

```
@@ -181,6 +181,7 @@
         <property name="MSBuildFrameworkToolsRoot" value="$(SystemRoot)\Microsoft.NET\Framework\" />
         <property name="SDK35ToolsPath" value="$([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\Windows\v8.0A\WinSDK-NetFx35Tools-x86', 'InstallationFolder', null, RegistryView.Registry32))" />
         <property name="SDK40ToolsPath" value="$([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8\WinSDK-NetFx40Tools-x86', 'InstallationFolder', null, RegistryView.Registry32))" />
+        <property name="SDK481ToolsPath" value="$([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8.1\WinSDK-NetFx40Tools-x86', 'InstallationFolder', null, RegistryView.Registry32))" />
```

Another option would avoid the need for downstream changes:

```suggestion
        <property name="SDK40ToolsPath" value="$([MSBuild]::ValueOrDefault($([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8\WinSDK-NetFx40Tools-x86', 'InstallationFolder', null, RegistryView.Registry32)), $([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8.1\WinSDK-NetFx40Tools-x86', 'InstallationFolder', null, RegistryView.Registry32))" />
```

Think that's better, or too complicated to be helpful?

---

This works, though I think it should be ValueOrDefault(<481 registry key>, <48 registry key>) so we can fall back to 48

---

Yes, definitely!

---

I verified this logic works and correctly falls back to 4.8 after I uninstalled 4.8.1.

(see "Foo: Test: ...")
```
C:\src\temp\11-15\9_36>msbuild
MSBuild version 17.5.0-preview-22519-01+82a30d307 for .NET Framework
Build started 11/15/2022 9:38:25 AM.
Project "C:\src\temp\11-15\9_36\proj.proj" on node 1 (default targets).
Foo:
 <SEE THIS>---->  Test: C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8.1 Tools\
...
[C:\src\temp\11-15\9_36\proj.proj]
CopyFilesToOutputDirectory:
  proj ->
Done Building Project "C:\src\temp\11-15\9_36\proj.proj" (default targets).


Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.44

C:\src\temp\11-15\9_36>msbuild
MSBuild version 17.5.0-preview-22519-01+82a30d307 for .NET Framework
Build started 11/15/2022 9:42:53 AM.
Project "C:\src\temp\11-15\9_36\proj.proj" on node 1 (default targets).
Foo:
<SEE THIS>---->  Test: C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\
...
 [C:\src\temp\11-15\9_36\proj.proj]
CopyFilesToOutputDirectory:
  proj ->
Done Building Project "C:\src\temp\11-15\9_36\proj.proj" (default targets).


Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.44
```

The project:
```xml
<Project Sdk="Microsoft.Build.NoTargets/3.1.0" InitialTargets="Foo">

    <PropertyGroup>
        <TargetFramework>net472</TargetFramework>
        <Test>$([MSBuild]::ValueOrDefault($([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8.1\WinSDK-NetFx40Tools-x86', 'InstallationFolder', null, RegistryView.Registry32)), $([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8\WinSDK-NetFx40Tools-x86', 'InstallationFolder', null, RegistryView.Registry32))))</Test>
    </PropertyGroup>

    <Target Name="Foo">
        <Message Text="Test: $(Test)" Importance="High" />
    </Target>

</Project>

```